src/elm/Pile.elm

```diff
@@ -0,0 +1,190 @@
+module Pile exposing (..)
+
+import Draggable
+import Draggable.Events
+import Html exposing (Html)
+import Html.Attributes
+import Html.Events
+import Math.Vector2 exposing (Vec2)
+
+
+type alias Id =
+    String
+
+
+type alias Card =
+    { id : Id
+    , position : Vec2
+    , cont : Html Msg
+    }
+
+
+dragCardBy : Vec2 -> Card -> Card
+dragCardBy delta card =
+    { card | position = card.position |> Math.Vector2.add delta }
+
+
+type alias CardPile =
+    { uid : Int
+    , activeCard : Maybe Card
+    , idleCards : List Card
+    }
+
+
+emptyPile : CardPile
+emptyPile =
+    CardPile 0 Nothing []
+
+
+addCard : ( Vec2, Html Msg ) -> CardPile -> CardPile
+addCard ( pos, cont ) ({ uid, idleCards } as pile) =
+    { pile
+        | uid = uid + 1
+        , idleCards = Card (String.fromInt uid) pos cont :: idleCards
+    }
+
+
+makeCardPile : List ( Vec2, Html Msg ) -> CardPile
+makeCardPile elements =
+    List.foldl addCard emptyPile elements
+
+
+allCards : CardPile -> List Card
+allCards { activeCard, idleCards } =
+    activeCard
+        |> Maybe.map (\c -> c :: idleCards)
+        |> Maybe.withDefault idleCards
+
+
+startDragging : Id -> CardPile -> CardPile
+startDragging id pack =
+    let
+        ( targetAsList, others ) =
+            List.partition (.id >> (==) id) (allCards pack)
+    in
+    { pack
+        | idleCards = others
+        , activeCard = targetAsList |> List.head
+    }
+
+
+stopDragging : CardPile -> CardPile
+stopDragging pack =
+    { pack | idleCards = allCards pack, activeCard = Nothing }
+
+
+dragActiveBy : Vec2 -> CardPile -> CardPile
+dragActiveBy delta ({ activeCard } as pack) =
+    { pack | activeCard = activeCard |> Maybe.map (dragCardBy delta) }
+
+
+type alias Model =
```

Comment:
I've so far resisted moving any state and commands out of the Main - but I think this is OK to do like this. Especially as it's already done! We may want to lift the Terminal out as well.

Reply:
I initially did this because I wanted to integrate two bits of unkown (to me)  code but having got to where this PR is now I'm not sure it was the best fit. Interesting though.